### ISSUE TITLE
[WebToolsE2E][Aspire] Run the project and open the URL in the browser, then close the browser, there is an error "Error loading data from the resource service" in CMD. 

### ISSUE THREAD

**Issue by Rita003** [dotnet/aspire#4768]
**REGRESSION INFO**: Work in Aspire 8.0.2

**INSTALL STEPS**

1. Clean machine: Win11 x64 23h2 ENU
2. Install SDK 8.0.302
3. Install Aspire 8.1.0-preview.1.24352.6

**REPRO STEPS**
1. Open CMD, create an Aspire App using following commands:
```
dotnet new aspire -o AspireApp
cd AspireApp/AspireApp.AppHost

```
2. Dotnet run the project in CMD, and open URL in the browser.
`dotnet run`
3. Close the browser

**ACTUAL**
There is an error in CMD
![Screenshot 2024-07-03 174255](https://github.com/dotnet/aspire/assets/88122107/2b531072-773a-44d5-b2b2-7a08f06d417c)

**EXPECTED**
There is not any error in CMD when close the browser

**Error Log**
```
fail: Aspire.Hosting.Dashboard.Model.DashboardClient[0]
      Error loading data from the resource service.
```

**Comment by balachir** [comment]
@kvenkatrajan I assigned this to area-dashboard with high pri for 8.1 as it's a regression in the user experience compared to 8.0. Can you triage and re-direct, please?

**Comment by kvenkatrajan** [comment]
@adamint  for initial investigation

**Comment by adamint** [comment]
Commit https://github.com/dotnet/aspire/commit/9039a55d9d7f9c5ad4f749b591c19e36c6710063#diff-87fafbca6bd03b2db55d5d2b4552a1bb72bdf02691f5f9b4e55e2795dfe0d995 added a try/catch - the exception thrown here is `OperationCanceledException`, @karolz-ms can that safely be ignored? Assigning to you temporarily.

**Comment by karolz-ms** [comment]
The previous version of the code was only awaiting on the task that gets the application information (a.k.a. `ConnectAsync`). Now we are also awaiting on the watch task. I think this is the reason why this error message is being shown now. 

On a cursory glance I would say the `OperationCanceledException` should not be logged, but just re-thrown. @JamesNK I believe you authored this part of https://github.com/dotnet/aspire/pull/4661 , what is your take?

**Comment by JamesNK** [comment]
Rethrown to where? This code is inside a task. Previously errors were unrecorded and unhandled.

The change here could be to not log certain errors in certain situations.

**Comment by karolz-ms** [comment]
Right, that is what I mean. Not log `OperationCanceledException` since it is not really an unexpected error, but just re-throw as it is currently done, here https://github.com/dotnet/aspire/blob/main/src/Aspire.Dashboard/ResourceService/DashboardClient.cs#L241 

One possibility is to just use an exception filter (`when ex is not OperationCanceledException`)

**Comment by Rita003** [comment]
Verify this issue fixed on the latest Aspire 8.1.0-preview.1.24360.6
